tests/SprykerEcoTest/Zed/Payone/Business/Payment/MethodMapper/KlarnaTest.php

```diff
@@ -0,0 +1,244 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEcoTest\Zed\Payone\Business\Payment\MethodMapper;
+
+use Generated\Shared\Transfer\PayoneKlarnaStartSessionRequestTransfer;
+use Generated\Shared\Transfer\QuoteTransfer;
+use SprykerEco\Zed\Payone\Business\Payment\MethodMapper\Klarna;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * @group Unit
+ * @group SprykerEco
+ * @group Zed
+ * @group Payone
+ * @group Business
+ * @group Payment
+ * @group MethodMapper
+ * @group KlarnaTest
+ */
+class KlarnaTest extends AbstractMethodMapperTest
+{
+    public const PAY_METHOD_TYPE = 'KIV';
+
+    protected const STANDARD_PARAMETER_CLEARING_TYPE = 'fnc';
+
+    public const AUTHORIZATION_KLARNA_REQUIRED_PARAMS = [
+        'financingtype' => self::PAY_METHOD_TYPE,
+    ];
+
+    protected const PREAUTHORIZATION_KLARNA_REQUIRED_PARAMS = [
+        'financingtype' => self::PAY_METHOD_TYPE,
+    ];
+
+    protected const PREAUTHORIZATION_COMMON_REQUIRED_PARAMS = [
+        'aid' => self::STANDARD_PARAMETER_AID,
+        'clearingtype' => self::STANDARD_PARAMETER_CLEARING_TYPE,
+        'reference' => self::PAYMENT_REFERENCE,
+        'amount' => self::AMOUNT_FULL,
+        'currency' => self::STANDARD_PARAMETER_CURRENCY,
+        'email' => self::DEFAULT_EMAIL,
+    ];
+
+    protected const AUTHORIZATION_COMMON_REQUIRED_PARAMS = [
+        'aid' => self::STANDARD_PARAMETER_AID,
+        'clearingtype' => self::STANDARD_PARAMETER_CLEARING_TYPE,
+        'reference' => self::PAYMENT_REFERENCE,
+        'amount' => self::AMOUNT_FULL,
+        'currency' => self::STANDARD_PARAMETER_CURRENCY,
+    ];
+
+    public const PREAUTHORIZATION_PERSONAL_DATA_REQUIRED_PARAMS = [
+        'lastname' => self::ADDRESS_LAST_NAME,
+        'country' => self::COUNTRY_ISO2CODE,
+        'language' => self::STANDARD_PARAMETER_LANGUAGE,
+        'email' => self::DEFAULT_EMAIL,
+    ];
+
+    public const START_SESSION_COMMON_REQUIRED_PARAMS = [
+        'aid' => self::STANDARD_PARAMETER_AID,
+        'clearingtype' => self::STANDARD_PARAMETER_CLEARING_TYPE,
+        'reference' => self::PAYMENT_REFERENCE,
+        'amount' => self::AMOUNT_FULL,
+        'currency' => self::STANDARD_PARAMETER_CURRENCY,
+    ];
+
+    public const START_SESSION_PERSONAL_DATA_REQUIRED_PARAMS = [
+        'lastname' => self::ADDRESS_LAST_NAME,
+        'country' => self::COUNTRY_ISO2CODE,
+        'language' => self::STANDARD_PARAMETER_LANGUAGE,
+        'email' => self::DEFAULT_EMAIL,
+    ];
+
+    /**
+     * @return void
+     */
+    public function testMapPaymentToPreauthorization()
+    {
+        $paymentEntity = $this->getPaymentEntityMock();
```

Comment:
All test methods should contains three blocks commented with:
// Arrange
// Act
// Assert

Reply:
Added